src/main/java/minijava/ir/optimize/AliasAnalyzer.java

```diff
@@ -151,27 +151,28 @@ private void updateMemory(Node node, Memory newMemory) {
   }
 
   @Override
+  public void defaultVisit(Node node) {
+    // We have to merge all preceding memory values
+    Memory memory =
+        seq(node.getPreds()).map(this::getMemory).foldLeft(Memory.empty(), Memory::mergeWith);
+    updateMemory(node, memory);
+  }
+
+  @Override
   public void visit(Phi node) {
-    if (node.getMode().equals(Mode.getM())) {
-      Memory memory =
-          seq(node.getPreds()).map(this::getMemory).foldLeft(Memory.empty(), Memory::mergeWith);
-      updateMemory(node, memory);
-    } else {
+    defaultVisit(node);
```

Comment:
I could probably pull out the logic of `defaultVisit` into `mergeMemoryFromPreds` and call that instead, sounds the like the most reasonable thing to do.